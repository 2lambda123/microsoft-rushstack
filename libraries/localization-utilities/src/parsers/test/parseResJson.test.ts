// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { parseResJson } from '../parseResJson';

describe(parseResJson.name, () => {
  it('parses a valid file', () => {
    const content: string = JSON.stringify({
      foo: 'Foo',
      '_foo.comment': 'A string'
    });

    expect(
      parseResJson({
        content,
        filePath: 'test.resjson'
      })
    ).toMatchSnapshot();
  });

  it('throws on excess comments', () => {
    const content: string = JSON.stringify({
      foo: 'Foo',
      '_bar.comment': 'A string'
    });

    expect(() =>
      parseResJson({
        content,
        filePath: 'test.resjson'
      })
    ).toThrowErrorMatchingSnapshot();
  });
});
