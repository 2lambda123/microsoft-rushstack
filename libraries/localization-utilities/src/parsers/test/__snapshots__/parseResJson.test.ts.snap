// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseResJson parses a valid file 1`] = `
Object {
  "foo": Object {
    "comment": "A string",
    "value": "Foo",
  },
}
`;

exports[`parseResJson throws on excess comments 1`] = `"The resjson file is invalid. Comments exist for the following string keys that don't have values: bar."`;
